ext/lb/haproxy/haproxy.go

```diff
@@ -43,7 +44,17 @@ func (p *HAProxyLoadBalancer) ConfigPath() string {
 }
 
 func (p *HAProxyLoadBalancer) Template() string {
-	return haproxyConfTemplate
+	d, err := ioutil.ReadFile(p.cfg.TemplatePath)
+
+	if err == nil {
+		return string(d)
+	} else {
+		log().Infof("Missing haproxy configuration template: file=%s", p.cfg.TemplatePath)
```

Comment:
We have this spread throughout.  We should not `log.Fatal` in the extensions.  This should be updated to return `(string, error)` and then bubble the error up.